provider/oci/images_test.go

```diff
@@ -0,0 +1,454 @@
+)
+
+func getShapeObject(shapeName string) ociCore.Shape {
+	return ociCore.Shape{
+		Shape: makeStringPointer(shapeName),
+	}
+}
+
+func makeShapesRequestResponse(compartment, id string, shapeNames []string) (ociCore.ListShapesRequest, ociCore.ListShapesResponse) {
+	shapesRequest := ociCore.ListShapesRequest{
+		CompartmentId: &compartment,
+		ImageId:       &id,
+	}
+
+	ociShapes := []ociCore.Shape{}
+	for _, val := range shapeNames {
+		ociShapes = append(ociShapes, getShapeObject(val))
+	}
+
+	shapesResponse := ociCore.ListShapesResponse{
+		Items: ociShapes,
+	}
+
+	return shapesRequest, shapesResponse
+}
+
+func makeListImageRequestResponse(compartment string, imgDetails []map[string]string) (ociCore.ListImagesRequest, ociCore.ListImagesResponse) {
+	ret := ociCore.ListImagesResponse{}
+	for _, val := range imgDetails {
+		id := makeStringPointer(val["id"])
+		name := makeStringPointer(val["name"])
+		os := makeStringPointer(val["os"])
+		version := makeStringPointer(val["version"])
+		ret.Items = append(
+			ret.Items,
+			ociCore.Image{
+				CompartmentId:          &compartment,
+				Id:                     id,
+				OperatingSystem:        os,
+				OperatingSystemVersion: version,
+				DisplayName:            name,
+			})
+	}
+
+	return ociCore.ListImagesRequest{CompartmentId: &compartment}, ret
+}
+
+type imagesSuite struct {
+	jujutesting.BaseSuite
+
+	testImageID     string
+	testCompartment string
+}
+
+var _ = gc.Suite(&imagesSuite{})
+
+func (i *imagesSuite) SetUpTest(c *gc.C) {
+	i.BaseSuite.SetUpTest(c)
+	oci.SetImageCache(&oci.ImageCache{})
+
+	i.testImageID = "ocid1.image.oc1.phx.aaaaaaaaa5mikpf5fktj4x47bx4p4ak4g5jyuyukkxpdg4nll36qzqwjzd2q"
+	i.testCompartment = "ocid1.compartment.oc1..aaaaaaaaakr75vvb5yx4nkm7ag7ekvluap7afa2y4zprswuprcnehqecwqga"
+}
+
+func (i *imagesSuite) TestNewImageVersion(c *gc.C) {
+	name := "Canonical-Ubuntu-14.04-2017.08.22-0"
+	img := ociCore.Image{
+		DisplayName: &name,
+	}
+	timeStamp, _ := time.Parse("2006.01.02", "2017.08.22")
+	version, err := oci.NewImageVersion(img)
+	c.Assert(err, gc.IsNil)
+	c.Assert(version.TimeStamp, gc.Equals, timeStamp)
+	c.Assert(version.Revision, gc.Equals, 0)
+}
+
+func (i *imagesSuite) TestNewImageVersionInvalidDate(c *gc.C) {
+	name := "Canonical-Ubuntu-14.04-NotARealDate-0"
+	img := ociCore.Image{
+		DisplayName: &name,
+	}
+	_, err := oci.NewImageVersion(img)
+	c.Assert(err, gc.ErrorMatches, "parsing time for.*")
+}
+
+func (i *imagesSuite) TestNewImageVersionInvalidRevision(c *gc.C) {
+	name := "Canonical-Ubuntu-14.04-2017.08.22-IShouldBeNumeric"
+	img := ociCore.Image{
+		DisplayName: &name,
+	}
+	_, err := oci.NewImageVersion(img)
+	c.Assert(err, gc.ErrorMatches, "parsing revision for.*")
+}
+
+func (i *imagesSuite) TestNewImageVersionInvalidName(c *gc.C) {
+	name := "fakeInvalidName"
+	img := ociCore.Image{
+		DisplayName: &name,
+	}
+	_, err := oci.NewImageVersion(img)
+	c.Assert(err, gc.ErrorMatches, "invalid image display name.*")
```

Comment:
it would be nicer to have this as `invalid image display name "fakeInvalidName"` so that we can see that the errors are useful to users.